enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/SdkIframeEmbedPreview.tsx

```diff
@@ -27,6 +30,22 @@ export const SdkIframeEmbedPreview = () => {
   const scriptRef = useRef<HTMLScriptElement | null>(null);
 
   const instanceUrl = useSetting("site-url");
+  const applicationColors = useSetting("application-colors");
+
+  // TODO: There is a bug in the SDK where if we set the theme back to undefined,
+  // some color will not be reset to the default (e.g. text color, CSS variables).
+  // We can remove this block once the bug is fixed.
+  const defaultTheme: MetabaseTheme = useMemo(() => {
+    const colors = Object.fromEntries(
+      getConfigurableThemeColors().map((color) => [
+        color.key,
+        applicationColors?.[color.originalColorKey] ??
+          defaultMetabaseColors[color.originalColorKey],
+      ]),
+    );
+
+    return { colors };
+  }, [applicationColors]);
```

Comment:
There is a bug I found where if you update the theme settings to `undefined` (i.e. `{ theme: undefined }`) using the `embed.updateSettings` method (which is in use by onAttributeChanged and the embed preview), some colors weirdly does not get reset:

- Text color
- Background of the Date Picker

I also noticed the CSS variables are not reset. I'm adding a workaround for now as this is an existing bug, and I feel it would take a while to fix.

<img width="2514" height="1766" alt="CleanShot 2568-08-06 at 02 31 45@2x" src="https://github.com/user-attachments/assets/cbbc7755-f2b6-4686-b694-f2e59cda5b07" />

Reply:
Created a Linear task for the above bug: https://linear.app/metabase/issue/EMB-696/when-updating-the-theme-back-to-undefined-in-embed-js-some-colors. We will have to check if this also happens on the SDK.